src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -150,30 +151,83 @@ def __init__(
         self.direct_deps_only = direct_deps_only
         self.description = description
 
-    @classmethod
-    def for_requirements(
-        cls,
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class _RelevantTargetsRequest:
```

Comment:
We use "AllTargets" to means something slightly different in other contexts, so to avoid confusion I went with "RelevantTargets" as the name of the thing we were referring to as `all_targets` in this file.